### DISCUSSION TITLE
How to get server to run on port 80 and 443

### DISCUSSION THREAD

**Discussion by officialbubies** [zoonk/uneebee#180] [Questions]
Hi, I am hosting this app on an AWS EC2 instance running Ubuntu. I have everything setup but cannot figure out how to get the server to be accessed on port 80 HTTP and 443 HTTPS. I have .pem files. I'm not sure if I need to change something in one of the /config .exs files or if I need to run a reverse proxy through Nginx etc. How is the appropriate way to do this? Thanks

**Comment by ceolinwill** [comment]
In theory, it should be working on port 443 for HTTPS because of [this line](https://github.com/zoonk/uneebee/blob/main/config/runtime.exs#L93). I'm running it on fly.io on port 443. To use port `80` for HTTP, you can set a `PORT` environment variable. It reads your environment variable [here](https://github.com/zoonk/uneebee/blob/main/config/runtime.exs#L90) or it uses `8080` as the default.

If you need to pass a custom keyfile/certfile, you might need to add a `https` key to the [config/runtime.exs file](https://github.com/zoonk/uneebee/blob/main/config/runtime.exs#L93). Something like this:

```elixir
https: [
  port: 443,
  cipher_suite: :strong,
  keyfile: System.get_env("SOME_APP_SSL_KEY_PATH"),
  certfile: System.get_env("SOME_APP_SSL_CERT_PATH")
]
```

That's the configuration I have for the [dev environment](https://github.com/zoonk/uneebee/blob/main/config/dev.exs#L23-L29).

Let me know if that works and/or if we need to make code changes to make it easier.
